lib/Platform/Intl/PlatformIntlApple.mm

```diff
@@ -337,48 +490,1307 @@ LocaleMatch lookupMatcher(
+  return nsStringToU16String(nstrTime);
+}
+static NSDictionary<NSString *, NSString *> *getDictOfTimeZoneNames() {
+  NSDictionary<NSString *, NSString *> *dictListOfTimeZones;
+  return dictListOfTimeZones = [NSTimeZone abbreviationDictionary];
+}
+bool isTimeZoneValid(const std::u16string &timeZoneValue) {
+  static NSDictionary<NSString *, NSString *> *dictListOfTimeZones =
+      getDictOfTimeZoneNames();
+  NSString *NSTimeValue = u16StringToNSString(timeZoneValue);
+  for (NSString *key in dictListOfTimeZones) {
+    if ([key isEqualToString:NSTimeValue]) {
+      return true;
+    }
+  }
+  return false;
+}
+// https://github.com/facebook/hermes/blob/33cf15ab318694423926c3762e001465b399ef38/lib/Platform/Intl/java/com/facebook/hermes/intl/PlatformDateTimeFormatterAndroid.java#L121
+std::u16string searchForDateFormatPattern(
+    const std::u16string &dateFormatPattern,
+    const std::vector<std::u16string> &values) {
+  if (dateFormatPattern.find(u"h") < dateFormatPattern.length()) {
+    return u"h12";
+  } else if (dateFormatPattern.find(u"K") < dateFormatPattern.length()) {
+    return u"h11";
+  } else if (dateFormatPattern.find(u"H") < dateFormatPattern.length()) {
+    return u"h23";
+  } else {
+    return u"h24";
+  }
+}
+std::u16string getDefaultHourCycle(NSLocale *locale) {
+  static std::u16string dateFormatPattern =
+      nsStringToU16String([NSDateFormatter dateFormatFromTemplate:@"j"
+                                                          options:0
+                                                           locale:locale]);
+  return searchForDateFormatPattern(dateFormatPattern, {u"h", u"K", u"H", u""});
+}
+// Implementation of
+// https://tc39.es/ecma402/#sec-initializedatetimeformat
 vm::ExecutionStatus DateTimeFormat::initialize(
     vm::Runtime *runtime,
     const std::vector<std::u16string> &locales,
     const Options &options) noexcept {
-  impl_->locale = u"en-US";
+  // 1. Let requestedLocales be ? CanonicalizeLocaleList(locales).
+  const vm::CallResult<std::vector<std::u16string>> requestedLocales =
+      getCanonicalLocales(runtime, locales);
+  if (LLVM_UNLIKELY(requestedLocales == vm::ExecutionStatus::EXCEPTION)) {
+    return vm::ExecutionStatus::EXCEPTION;
+  }
+  // 2. Let options be ? ToDateTimeOptions(options, "any", "date").
+  auto optionsTo = toDateTimeOptions(options, u"any", u"date", runtime);
+  if (optionsTo == vm::ExecutionStatus::EXCEPTION) {
+    return vm::ExecutionStatus::EXCEPTION;
+  }
+  auto optionsToUse = optionsTo.getValue();
+  // 3. Let opt be a new Record.
+  Impl::newRecord opt;
+  // 4. Let matcher be ? GetOption(options, "localeMatcher", "string", «
+  //   "lookup", "best fit" », "best fit").
+  auto matcher = getOptionString(
+      optionsToUse,
+      u"localeMatcher",
+      {u"lookup", u"best fit"},
+      llvh::Optional<std::u16string>(u"best fit"),
+      runtime);
+  // 5. Set opt.[[localeMatcher]] to matcher.
+  opt.localeMatcher = matcher->getValue();
```

Comment:
I didn't check for hasValue on this one because the fallback is "best fit" (normally it's blank), so in theory it has to have a value. But is it safer to just use hasValue() anyway?

Reply:
Sorry, I missed that there is a fallback value, thank yo for the clarification.